### PR TITLE
Storage validation tests cleanup

### LINKED ISSUE #99005: Use helper functions to clean up test-cases (specifically net but not really specific)
Inspired by #98726

There are a lot of tests (pkg/apis/networking/validation/validation_test.go in that PR) that have tables of structures like Service or NetworkPolicy, where the struct has to be complete enough (e.g. to pass validation) but each test-case only modifies one tiny section of the otherwise boilerplate struct.

The result is a LOT of noise which makes the actual differences hard to see.

Compare to something like pkg/registry/core/service/strategy_test.go `TestDropTypeDependentFields`, where all the boilerplate is wrapped up in a helper function, and specifically named "tweaks" are run against the result.  The net result is that the test-case only emphasizes things it is changing and is WAY shorter and less noisy.

I'm filing this because I look at a lot of sig-net tests, but it can apply anywhere we have repeated struct initializers eating tons of LOC.

This pattern can be applied incrementally - one test function or file at a time.  Good opportunity to rack up some cleanup PRs.

### PR DESCRIPTION
**Pull request by mowangdk**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Boilerplcate Storage related struct and a helper function that applies tweaks to create specific test cases.

#### Which issue(s) this PR fixes:

refs #99005 

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?

None

### CODE REVIEW

**Comment by JornShen** [comment]
Can you split it to two commit? One commit change one test case. This will be more clear. 
:)

**Comment by mowangdk** [comment]
> Can you split it to two commit? One commit change one test case. This will be more clear.
> :)

I would like to put the entire file modification into one commit, current commit message will be modified in next submit 😆

**Comment by mowangdk** [comment]
@JornShen @deads2k @msau42 PTAL

**Comment by mowangdk** [comment]
> /ok-to-test

ptal~

**Comment by JornShen** [comment]
you can send msg to sig-storage in slack(k8s.io) to raise more attention.

File: pkg/apis/storage/validation/validation_test.go [hunk 1]
```diff
@@ -54,50 +54,53 @@ var (
...
+		}
+	}
+	setRetrainReClaimPolicyElement := func(storageClass *storage.StorageClass) {
+		storageClass.ReclaimPolicy = &retainReclaimPolicy
+	}
+	successCases := []*storage.StorageClass{
+		makeStorageClassCustom(setEmptyParametersElement, setImmediateVolumeBindingModeElement),
+		makeStorageClassCustom(setImmediateVolumeBindingModeElement),
+		makeStorageClassCustom(setMultiParametersElement, setImmediateVolumeBindingModeElement),
+		makeStorageClassCustom(setRetrainReClaimPolicyElement, setImmediateVolumeBindingModeElement),
```
Comment (pohly) [inline pkg/apis/storage/validation/validation_test.go:98]: I find the `*Element` suffix redundant.

Comment (mowangdk) [reply pkg/apis/storage/validation/validation_test.go:98]: ok, I will remove these *Element suffixes~

File: pkg/apis/storage/validation/validation_test.go [hunk 2]
```diff
@@ -111,309 +114,190 @@ func TestValidateStorageClass(t *testing.T) {
...
-			Provisioner:   "kubernetes.io/foo",
-			ReclaimPolicy: &recycleReclaimPolicy,
-		},
+	errorCases := map[string]*storage.StorageClass{
+		"namespace is present":         makeStorageClassCustom(setNamespaceIsPresentElement),
+		"invalid provisioner":          makeStorageClassCustom(setInvalidProvisionerElement),
+		"invalid empty parameter name": makeStorageClassCustom(setEmptyParameterNameElement),
+		"provisioner: Required value":  makeStorageClassCustom(setEmptyProvisionerElement),
+		"too long parameters":          makeStorageClassCustom(setTooLongParametersElement),
+		"invalid reclaimpolicy":        makeStorageClassCustom(setInvalidReclaimPolicyElement),
```
[the remark this answers is not in the document]
Comment (mowangdk) [reply pkg/apis/storage/validation/validation_test.go:142]: get

**Comment by mowangdk** [comment]
> owever our team prefers the original over the proposed because it's simply easier to read and follow because the object is small enough.

thanks for the comment. IMHO this proposal is very meaningful here ~

**Comment by mowangdk** [comment]
@pohly ptal, I have modified the code according to your suggestion~

**Comment by pohly** [comment]
I'll let @mauriciopoppe decide whether this is considered more readable than before.

**Comment by msau42** [comment]
Thank you @mowangdk for the contribution. However I agree with @mauriciopoppe that in this case, most of the test objects are small enough and we're changing so many different fields in our tests that applying this pattern makes the tests harder to read. There are definitely situations where this pattern would greatly improve readability, but I don't think this is one of them.

**Comment by mowangdk** [comment]
> Thank you @mowangdk for the contribution. However I agree with @mauriciopoppe that in this case, most of the test objects are small enough and we're changing so many different fields in our tests that applying this pattern makes the tests harder to read. There are definitely situations where this pattern would greatly improve readability, but I don't think this is one of them.

ok, get the point, I will close this pr

**Pull request closed without merging**